Sources/PrimerSDK/Classes/Core/Checkout Components/PrimerCheckoutComponents.swift

```diff
@@ -0,0 +1,222 @@
+//
+//  PRTextField.swift
+//  PrimerSDK
+//
+//  Created by Evangelos on 28/1/22.
+//
+
+#if canImport(UIKit)
+
+import UIKit
+
+public class PrimerCheckoutComponents {
+    
+    public static var delegate: PrimerCheckoutComponentsDelegate?
+    
+    public static func listAvailablePaymentMethods(forSession clientToken: String, completion: @escaping ([PaymentMethodConfigType]?, Error?) -> Void) {
+        do {
+            try ClientTokenService.storeClientToken(clientToken)
+        } catch {
+            completion(nil, error)
+        }
+        
+        let primerConfigurationService: PaymentMethodConfigServiceProtocol = DependencyContainer.resolve()
+        firstly {
+            primerConfigurationService.fetchConfig()
+        }
+        .done {
+            let availablePaymentMethodTypes = PrimerConfiguration.paymentMethodConfigs?.compactMap({ $0.type })
+            completion(availablePaymentMethodTypes, nil)
+        }
+        .catch { err in
+            completion(nil, err)
+        }
+    }
+    
+    public static func listInputElementTypes(for paymentMethodType: PaymentMethodConfigType) -> [PrimerInputElementType] {
+        switch paymentMethodType {
+        case .adyenAlipay:
+            return []
+        case .adyenDotPay:
+            return []
+        case .adyenGiropay:
+            return []
+        case .adyenIDeal:
+            return []
+        case .adyenMobilePay:
+            return []
+        case .adyenSofort:
+            return []
+        case .adyenTrustly:
+            return []
+        case .adyenTwint:
+            return []
+        case .adyenVipps:
+            return []
+        case .apaya:
+            return []
+        case .applePay:
+            return []
+        case .atome:
+            return []
+        case .buckarooBancontact:
+            return []
+        case .buckarooEps:
+            return []
+        case .buckarooGiropay:
+            return []
+        case .buckarooIdeal:
+            return []
+        case .buckarooSofort:
+            return []
+        case .goCardlessMandate:
+            return []
+        case .googlePay:
+            return []
+        case .hoolah:
+            return []
+        case .klarna:
+            return []
+        case .mollieBankcontact:
+            return []
+        case .mollieIdeal:
+            return []
+        case .payNLBancontact:
+            return []
+        case .payNLGiropay:
+            return []
+        case .payNLIdeal:
+            return []
+        case .payNLPayconiq:
+            return []
+        case .paymentCard:
+            return [.cardNumber, .expiryDate, .cvv, .cardholderName]
+        case .payPal:
+            return []
+        case .xfers:
+            return []
+        case .other(let rawValue):
+            return []
+        }
+    }
+    
+    public static func getButton(for paymentMethodType: PaymentMethodConfigType) -> UIButton? {
```

Comment:
Same as above as precondition 😅 
----
This looks like a "factory method", so unless defined in our Primer's convention and styles (e.g. "We at primer write API with the same method signature among platform"), I guess here we could use:

```suggestion
    public static func makeButton(paymentMethodType: PaymentMethodConfigType) -> UIButton? {
```

Reply:
Done!